docs/dual-pods.md

```diff
@@ -159,10 +216,59 @@ spec:
       claimName: somenode-local
 ```
 
-Explicitly specifying a quantity of "0" GPUs gets this container
-access to all of the GPUs. Setting the `CUDA_VISIBLE_DEVICES` envar
-directs the `vllm serve` process to use the indicated one.
+### The dual-pods controller data and logic
+
+The mutable internal state of the controller includes the following.
+
+- A set of existing vLLM instances. Each runs a particular model, with
+  other command line parameters and some environment variable
+  settings. Each instance is in a Pod, on one Node, and uses a set of
+  particular accelerators on that Node.
+
+- A set of server-running Pods. Each is running one of the
+  aforementioned vLLM instances.
+
+- A set of server-requesting Pods that are bound to Nodes. Each such
+  Pod specifies: model, other command-line parameters, some
+  environment variable settings, and a Node. After its stub has been
+  queried, this Pod is also known to specify a particular set of
+  accelerators on the Node. This server-requesting Pod may be bound
+  (here, in this data structure) to a vLLM instance.
+
+When the server-requesting Pod is bound to a Node that is absent or in the process of being deleted, the dual-pods controller has nothing left to do and the following logic is irrelevant.
+When a server-requesting Pod is bound to a server-running Pod that is in the process of being deleted, the controller (a) ensures that its finalizer is not on the server-running Pod and (b) ensures that the server-requesting Pod is being deleted. (The controller creates server-running Pods with its finalizer on them, so that they cannot evaporate without this interaction with the controller.)
```

Comment:
Can we add another line between the different "When" scenarios so they don't all get combined into one big para?

Reply:
Sorry, the github diff display for the latest commit does not make it clear, but the commit only adds whitespace.